svc-update/update/getresources.go

```diff
@@ -87,3 +95,161 @@ func GetUpdateService() response.RPC {
 	return resp
 
 }
+
+// GetAllFirmwareInventory is a functioanlity to retrive all the available inventory
+// resources from the added BMC's
+func (e *ExternalInterface) GetAllFirmwareInventory(req *updateproto.UpdateRequest) response.RPC {
+	var resp response.RPC
+	resp.Header = map[string]string{
+		"Allow":             `"GET"`,
+		"Cache-Control":     "no-cache",
+		"Connection":        "keep-alive",
+		"Content-type":      "application/json; charset=utf-8",
+		"Transfer-Encoding": "chunked",
+		"OData-Version":     "4.0",
+	}
+	firmwareCollection := uresponse.Collection{
+		OdataContext: "/redfish/v1/$metadata#FirmwareInventoryCollection.FirmwareCollection",
+		OdataID:      "/redfish/v1/UpdateService/FirmwareInventory",
+		OdataType:    "#FirmwareInventoryCollection.FirmwareInventoryCollection",
+		Description:  "FirmwareInventory view",
+		Name:         "FirmwareInventory",
+	}
+	var members []dmtf.Link
+
+	firmwareCollectionKeysArray, err := e.DB.GetAllKeysFromTable("FirmwareInventory")
+	if err != nil || len(firmwareCollectionKeysArray) == 0 {
+		log.Printf("odimra Doesnt have Servers")
+	}
+
+	for _, key := range firmwareCollectionKeysArray {
+		members = append(members, dmtf.Link{Oid: key})
+	}
+	firmwareCollection.Members = members
+	firmwareCollection.MembersCount = len(members)
+	resp.Body = firmwareCollection
+	resp.StatusCode = http.StatusOK
+	return resp
+}
+
+// GetFirmwareInventory is used to fetch resource data. The function is supposed to be used as part of RPC
+// For getting firmware inventory resource information,  parameters need to be passed Request .
+// Request holds the  Uuid and Url ,
+// Url will be parsed from that search key will created
+// There will be two return values for the fuction. One is the RPC response, which contains the
+// status code, status message, headers and body and the second value is error.
+func (e *ExternalInterface) GetFirmwareInventory(req *updateproto.UpdateRequest) response.RPC {
+	var resp response.RPC
+	resp.Header = map[string]string{
+		"Allow":             `"GET"`,
+		"Cache-Control":     "no-cache",
+		"Connection":        "keep-alive",
+		"Content-type":      "application/json; charset=utf-8",
+		"Transfer-Encoding": "chunked",
+		"OData-Version":     "4.0",
+	}
+
+	requestData := strings.Split(req.ResourceID, ":")
+	if len(requestData) <= 1 {
+		errorMessage := "error: SystemUUID not found"
+		return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, errorMessage, []interface{}{"FirmwareInventory", req.ResourceID}, nil)
+	}
+	data, gerr := e.DB.GetResource("FirmwareInventory", req.URL)
+	if gerr != nil {
+		log.Printf("error getting firmware inventory details : %v", gerr.Error())
+		errorMessage := gerr.Error()
+		if errors.DBKeyNotFound == gerr.ErrNo() {
```

Comment:
While getting the System, Chassis or Manager Resource, if it is not found in DB the  resource, will be fetched from the device and stored again please confirm once whether that logic needs to be added here

Reply:
Addressed